### PR TITLE
Baseline chart model crudl

### PR DESCRIPTION
**Pull request by emullaney**
This is the Model and CRUDL for the new BaselineTerm object and new baseline app.

### CODE REVIEW

File: tracpro/static/libs/django-smart-selects/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Copyright (c) 2009, Patrick Lauber
```
Comment (rebecca-caktus) [inline tracpro/static/libs/django-smart-selects/LICENSE.txt]: Hmm, is this file needed?

File: tracpro/templates/frame.haml [hunk 2]
```diff
@@ -39,6 +39,10 @@
     - block extra-less
     - block extra-style
 
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
+    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.4/jquery-ui.min.js"></script>
```
Comment (rebecca-caktus) [inline tracpro/templates/frame.haml]: Should jquery ui need to be loaded on every page? I think we're only using it on a couple of them.

File: tracpro/static/libs/django-smart-selects/LICENSE.txt (hunk 1 above)
Comment (emullaney) [inline tracpro/static/libs/django-smart-selects/LICENSE.txt]: It was unclear to me, but re-reading the license now, maybe not...

File: tracpro/baseline/migrations/0005_auto_20150817_1547.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
```
Comment (rebecca-caktus) [inline tracpro/baseline/migrations/0005_auto_20150817_1547.py]: I think that these 5 migrations can be squashed into one initial migration. 

(I would do it by deleting all `baseline` migrations and re-running `django-admin makemigrations baseline` to create a brand new initial migration. However [squashmigrations](https://docs.djangoproject.com/en/1.8/topics/migrations/#squashing-migrations) looks like it does essentially the same thing.)

File: tracpro/baseline/forms.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            self.fields['baseline_poll'].queryset = polls
+            self.fields['follow_up_poll'].queryset = polls
+
+    def clean(self, *args, **kwargs):
+        cleaned_data = super(BaselineTermForm, self).clean()
+
+        start_date = cleaned_data.get("start_date")
+        end_date = cleaned_data.get("end_date")
+
+        if start_date > end_date:
```
Comment (rebecca-caktus) [inline tracpro/baseline/forms.py]: It is possible that one or both of `start_date` or `end_date` could be `None`. `None > None` works but you get a `TypeError` if you try to compare a `datetime` with None.

You can change the check to `start_date and end_date and start_date > end_date` to cover the error condition while allowing the default `This field is required` messages to suffice if one or both fields is `None`.

File: tracpro/baseline/forms.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
+from django import forms
+
+from tracpro.polls.models import Poll
+from .models import BaselineTerm
+
+
+class BaselineTermForm(forms.ModelForm):
+    """
+    Form for Baseline Term
+    """
```
Comment (rebecca-caktus) [inline tracpro/baseline/forms.py]: After our talk yesterday, I'm not sure whether `region` will remain in this form. But if it does (here or elsewhere), `django-mptt` has some useful tools for forms: http://django-mptt.github.io/django-mptt/forms.html

Comment (rebecca-caktus) [inline tracpro/baseline/forms.py]: Would it be difficult to use javascript to hide the `Baseline Question` and `Follow up Question` fields until the corresponding `Poll` field has data?

**Comment by rebecca-caktus** [comment]
Looks great so far!

File: tracpro/templates/frame.haml (hunk 2 above)
Comment (emullaney) [inline tracpro/templates/frame.haml]: I've moved it out of frame into baseline templates.

File: tracpro/baseline/forms.py (hunk 5 above)
Comment (emullaney) [inline tracpro/baseline/forms.py]: I've made this adjustment, please take a look at static\js\baseline.js and test it out.

Comment (emullaney) [inline tracpro/baseline/forms.py]: Also, I've remove region from the model for now.

File: tracpro/baseline/forms.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+        if start_date and end_date and start_date > end_date:
+            raise forms.ValidationError(
+                "Start date should be before end date."
+            )
+
+        baseline_question = cleaned_data.get("baseline_question")
+        follow_up_question = cleaned_data.get("follow_up_question")
+
+        if baseline_question and follow_up_question and baseline_question.pk == follow_up_question.pk:
```
Comment (rebecca-caktus) [inline tracpro/baseline/forms.py]: Django's [model equality implementation](https://github.com/django/django/blob/1.8.4/django/db/models/base.py#L506) essentially compares pks - so you can use `baseline_question == follow_up_question` here.

File: tracpro/static/js/baseline.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+$(function() {
+        $("input[class^='datepicker']").datepicker();
```
Comment (rebecca-caktus) [inline tracpro/static/js/baseline.js]: All but the first line are tabbed over once too many.

File: tracpro/static/js/baseline.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+          if(this.value) {
+            $("#id_baseline_question").parent().parent().show();
+          }
+          else {
+            $("#id_baseline_question").parent().parent().hide();
+          }
+        });
+
+        // Display/hide question if poll is selected
+        if (!$("#id_follow_up_poll").val()) {
```
Comment (rebecca-caktus) [inline tracpro/static/js/baseline.js]: You can remove some duplicated logic by calling `$("#id_follow_up_poll,#id_baseline_poll").change()` after hooking up the `'change'` behavior.

File: tracpro/static/js/baseline.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+            $("#id_follow_up_question").parent().parent().hide();
+        }
+        $("#id_follow_up_poll").on("change", function() {
+          if(this.value) {
```
Comment (rebecca-caktus) [inline tracpro/static/js/baseline.js]: This should be `$(this).val()`.

File: tracpro/static/js/baseline.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+            $("#id_follow_up_question").parent().parent().show();
```
Comment (rebecca-caktus) [inline tracpro/static/js/baseline.js]: What do you think about using `.closest(".form-group")`?

File: tracpro/templates/frame.haml [hunk 11]
```diff
@@ -114,22 +116,31 @@
                         %a{ href:"{% url 'msgs.message_list' %}" }
                           - trans "Message Log"
 
+              %li.dropdown
```
Comment (rebecca-caktus) [inline tracpro/templates/frame.haml]: I see the "Management" menu when I am logged out, but it is empty.

**Comment by rebecca-caktus** [comment]
Nifty `django-smart-selects` feature: If the poll only has one question, it is auto-selected!

File: tracpro/static/js/baseline.js (hunk 10 above)
Comment (emullaney) [inline tracpro/static/js/baseline.js]: Way better... thanks!

File: tracpro/static/js/baseline.js (hunk 8 above)
Comment (emullaney) [inline tracpro/static/js/baseline.js]: Awesome idea.

**Comment by rebecca-caktus** [comment]
One comment then :zap:
